fetools-app/src/components/ColorGradientSlider.jsx

```diff
@@ -0,0 +1,209 @@
+import { useState } from "react"
+import tinycolor from "tinycolor2";
+
+const colors = [
+    getRandomColor(), getRandomColor()
+]
+
+export default function ColorGradientSlider(){
+
+    const colorStyles = getCSSRules(':root')
+
+    const [,setActiveKnob] = useState(false)
+
+    const [updateColors, setUpdateColors] = useState(colors)
+
+    updateCSSValues('--color1-thumb', colors[0])
```

Comment:
Rather than updating CSS variables hard coded in index.css, it might be preferable to compose css code as a string and dynamically append it to the DOM, especially as we are going to compose that css string any in the code block displayed to end users.

Here's an example of this coding pattern:
https://medium.com/@vaibhav_55744/how-to-append-css-strings-dynamically-via-javascript-30c4ed4b6cf3

One concern here might be performance; I'm not sure if doing what I recommend here might result in unacceptable lag. With this in mind, you might take a hybrid approach where you update the style property of the preview box and slider directly. Whatever you do, you will need to generate the stops programmatically; hard coded variables will not suffice.